src/apiV2/types.ts

```diff
@@ -73,23 +73,16 @@ export type StreamQueryConfig = {
       startTag?: string;
       endTag?: string;
     };
-    reranker?:
-      | { type: "none" }
-      | {
-          type: "customer_reranker";
-          rerankerId: string;
-        }
-      | {
-          type: "mmr";
-          // Diversity bias ranges from 0 to 1.
-          // 0 will optimize for results that are as closely related to the query as possible.
-          // 1 will optimize for results that are as diverse as possible.
-          diversityBias: number;
-        }
-      | {
-          type: "userfn";
-          userFunction?: string;
-        };
```

Comment:
Let's keep the union type here and just add one of type chain that accepts one or more of the above. That would be closer to [what's in the docs](https://docs.vectara.com/docs/learn/chain-reranker). I think you can just add [this](https://github.com/vectara/stream-query-client/pull/33/files#diff-47efe2d131e70c383d345f42f43fcdb55963d00d60b58f351289b4c0fed01577R21) to this union type, and the type property of it could just be "chain".

Then the configuration passed into stream query client would be closer to what's in the chain reranker page:

```
// We'd be able to pass this in nearly verbatim as part of the stream query config
{
  "reranker": {
    "type": "chain",
    "rerankers": [
      {
        "type": "customer_specific",
        "reranker_id": "rnk_272725719"
      },
      {
        "type": "userfn",
        "user_function": "get('$.metadata.popularity') * get('$.metadata.score')"
      }
    ]
  }
}
```

It would also address the issue I previously mentioned where in the changes you made, users would need to remember the names in [this piece of code](https://github.com/vectara/stream-query-client/pull/33/files#diff-91282dbe5e13c57855620882866ebec7e14db63de85011f64792c0bcd259a458R17).